lib/spack/spack/extensions.py

```diff
@@ -24,39 +31,47 @@ def extension_name(path):
         path (str): path where the extension resides
 
     Returns:
-        The extension name or None if path doesn't match the format
-        for Spack's extension.
+        The extension name.
+
+    Raises:
+         ExtensionNamingError: if path does not match the expected format
+             for a Spack command extension.
     """
-    regexp_match = re.search(extension_regexp, os.path.basename(path))
+    regexp_match = re.search(_extension_regexp,
+                             os.path.basename(path.rstrip(os.sep)))
```

Comment:
@alalazo maybe:
```Python
    os.path.basename(os.path.normpath(os.path.realpath(path)))
```  
would work for you while avoiding the possible semantic change?